src/parser/mage/fire/modules/features/CombustionSpellUsage.js

```diff
@@ -14,22 +15,43 @@ class CombustionSpellUsage extends Analyzer {
     abilityTracker: AbilityTracker,
   };
 
-  castedWithInstants = 0;
+  badCastsStarted = 0;
+  badCastsCompleted = 0;
 
-  //Check to see if the player started casting Fireball or Scorch during Combustion while they had Fire Blast or Phoenix Flames charges available.
-  on_byPlayer_begincast(event) {
-    const spellId = event.ability.guid;
-    if ((spellId !== SPELLS.FIREBALL.id && spellId !== SPELLS.SCORCH.id) || !this.selectedCombatant.hasBuff(SPELLS.COMBUSTION.id)) {
+  constructor(...args) {
+    super(...args);
+    this.hasBlasterMaster = this.selectedCombatant.hasTrait(SPELLS.BLASTER_MASTER.id);
+    this.hasPhoenixFlames = this.selectedCombatant.hasTalent(SPELLS.PHOENIX_FLAMES_TALENT.id);
+    this.addEventListener(Events.begincast.by(SELECTED_PLAYER).spell([SPELLS.FIREBALL,SPELLS.SCORCH]), this.checkChargeCounts);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.FIREBALL,SPELLS.SCORCH]), this.checkChargeCounts);
+  }
+
+  //Check to see if the player cast (or started to cast) Fireball or Scorch during Combustion while they had Fire Blast or Phoenix Flames charges available.
+  checkChargeCounts(event) {
+    const hasCombustion = this.selectedCombatant.hasBuff(SPELLS.COMBUSTION.id);
+    const fireBlastCharges = this.spellUsable.chargesAvailable(SPELLS.FIRE_BLAST.id);
+    const phoenixFlamesCharges = (this.spellUsable.chargesAvailable(SPELLS.PHOENIX_FLAMES_TALENT.id) || 0);
+    //If the player has the Blaster Master trait, it is acceptable to cast Scorch during Combustion
+    if (!hasCombustion || (this.hasBlasterMaster && event.ability.guid === SPELLS.SCORCH.id)) {
       return;
     }
-    if (this.spellUsable.chargesAvailable(SPELLS.FIRE_BLAST.id) > 0 || this.spellUsable.chargesAvailable(SPELLS.PHOENIX_FLAMES_TALENT.id) > 0) {
-      this.castedWithInstants += 1;
-      debug && this.log("Casted with Instants Available");
+
+    if (event.type === "cast" && (fireBlastCharges > 0 || phoenixFlamesCharges > 0)) {
```

Comment:
You have two instances of ` && (fireBlastCharges > 0 || phoenixFlamesCharges > 0)`

Instead should probably (before the rest), just do `if(fireBlastCharges < 1  && phoenixFlamesCharges < 1) { return; }`